src/Mermaid/Mermaid.tsx

```diff
@@ -15,11 +15,29 @@
  */
 
 import { useEffect } from 'react';
+import { PaletteType, useTheme } from '@material-ui/core';
+
 import { useShadowRootElements } from '@backstage/plugin-techdocs-react';
 import mermaid from 'mermaid'
 import { isMermaidCode } from './hooks';
 import { MermaidProps } from './props';
+import { BackstageTheme } from '@backstage/theme';
+import mermaidAPI from 'mermaid/mermaidAPI';
+
+
+function selectConfig(backstagePalette: PaletteType, properties: MermaidProps): mermaidAPI.Config {
+  // Theme set directly in the Mermaid configuration takes
+  // precedence for backwards-compatibility
+  if(properties.config) {
+    return properties.config;
+  }
 
+  if(backstagePalette == 'light') {
+    return properties.lightConfig || {};
```

Comment:
I wonder whether we should really have different configs here vs. just different `theme / themeCSS`.

E.g. have functions like:

```js
function selectConfig(backstagePalette: PaletteType, properties: MermaidProps): mermaidAPI.Config {
  // Theme set directly in the Mermaid configuration takes
  // precedence for backwards-compatibility
  if(properties.theme) {
    return properties.theme;
  }
 
 if(backstagePalette == 'light') {
  return properties.lightTheme;
 }
 
  // etc.
 ```

As configs like fonts and padding should probably be shared across both modes. wdyt?

Reply:
I was wondering about this too, and decided to go with the "entire config" approach because it seems a bit more versatile. I'm not super familiar with MermaidJS yet, but looking at [the config](https://mermaid-js.github.io/mermaid/#/./Setup?id=mermaidapi-configuration-defaults) I can imagine that some of the parameters, like padding, margin, fill or font-size might be changed based on the theme to improve readability with specific color schemes.

Since it would be relatively easy to manage the common part of the configuration outside of mermaid, e.g. as in the code snippet below, I thought this might be a better approach than trying to predict what people _may_ want to configure in the future.

```js
const commonConfig = { [...] };
const lightConfig = { theme: 'base', ...commonConfig };
const darkConfig = { theme: 'dark', ...commonConfig };
```

Let me know what you think :)

Reply:
Yes, I was thinking about the same. Let go with this approach then 👍

Reply:
Nice 🚀 Anything that needs changing? Should I see if I can come up with a useful test for this behaviour?

Reply:
If you can think of a test, sure, but wouldn't really know what to test in this case as its probably difficult to derive the config from the rendered chart.

I guess you could add a unit test that the right config is picked 🤔

Reply:
I added a unit test. I had hoped to be able to add a component test, but, as far as I could tell, that would've required to get rid of `TechDocsAddonTester`, which would've been a larger change 🤷🏼‍♂️

Reply:
Thanks a lot, I haven't done much backstage development so any help here is appreciated :-)